app/src/main/java/com/zulip/android/models/Message.java

```diff
@@ -104,7 +105,7 @@
     private String subject;
     @SerializedName("timestamp")
     @DatabaseField(columnName = TIMESTAMP_FIELD)
-    private Date timestamp;
+    private long timestamp;
```

Comment:
Let's take a example (assume that back dated messages are stored in local DB (will be fine once https://github.com/zulip/zulip/issues/3406 is done) and today's date is  25/01/2017)

Now when we narrow to date let's say 20/01/2017. 
**Before this PR**
Database query is done with `where.like(Message.TIMESTAMP_FIELD, new SelectArg(date));` (https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/filters/NarrowFilterByDate.java#L59) rather than filtering messages of that specific date it just do nothing. When I debugged I found that list contains messages of all dates, but messages of other days ( except 20/01/2017) are not shown because it returns false on match (https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/filters/NarrowFilterByDate.java#L65). All DB quires are done with limit of 100. 
**Case 1**
So if messages from 20 to 25 are less than 100 thus messages returned by DB query includes messages of 20 and only messages dated 20 are shown with the help of match method. 
**Case 2**
Messages from 20 to 25 are greater than 100 thus messages returned by DB query does not include messages of 20, even if they are stored in the local DB. Also no messages are displayed so scrolling is not possible and no further DB quires takes place.

**Problem** Where clause is not working as per expectation.

**In this PR**
Changed where clause (https://github.com/vishwesh3/zulip-android/blob/narrow/app/src/main/java/com/zulip/android/filters/NarrowFilterByDate.java#L57) 
Now DB query returns only those messages whose timestamp is in between narrowed date.